packages/@aws-cdk/custom-resource-handlers/lib/aws-ecr/auto-delete-images-handler/index.ts

```diff
@@ -22,16 +23,14 @@ export async function autoDeleteHandler(event: AWSLambda.CloudFormationCustomRes
 async function onUpdate(event: AWSLambda.CloudFormationCustomResourceEvent) {
   const updateEvent = event as AWSLambda.CloudFormationCustomResourceUpdateEvent;
   const oldRepositoryName = updateEvent.OldResourceProperties?.RepositoryName;
-  const newRepositoryName = updateEvent.ResourceProperties?.RepositoryName;
-  const repositoryNameHasChanged = (newRepositoryName && oldRepositoryName)
-    && (newRepositoryName !== oldRepositoryName);
-
-  /* If the name of the repository has changed, CloudFormation will try to delete the repository
-     and create a new one with the new name. So we have to delete the images in the
-     repository so that this operation does not fail. */
-  if (repositoryNameHasChanged) {
-    return onDelete(oldRepositoryName);
-  }
+  const newRepositoryName = updateEvent.ResourceProperties?.RepositoryName ?? oldRepositoryName;
+
+  /* If the name of the repository has changed, CloudFormation will try to delete the repo
+  and create a new one with the new name. Returning a PhysicalResourceId that differs
+  from the event's PhysicalResourceId will trigger a `Delete` event for the custom
+  resource. The `Delete` event will trigger `onDelete` function which will
+  empty the content of the repository and then proceed to delete the repository. */
+  return { PhysicalResourceId: newRepositoryName };
```

Comment:
What was the original return value from `onDelete()` and how was it used after this function call? Or does the return value not matter and it was just to perform the deletion as part of thie `onUpdate()`? 

Also where is the logic that checks if the repository name has changed or not and triggers the deletion of the bucket at the end of the deployment?

Reply:
`onDelete` doesn't return any value so it doesn't matter. We used to directly call the delete methods defined in `onDelete` not not using the default behaviour CFN provided (which is to do the deletion for us if the id changes).

That logic is default with in CFN custom resource. `The value returned for a PhysicalResourceId can change custom resource update operations. If the value returned is the same, it is considered a normal update. If the value returned is different, AWS CloudFormation recognizes the update as a replacement and sends a delete request to the old resource. `. https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/crpg-ref-requesttypes-delete.html